src/components/CommentCard.jsx

```diff
@@ -1,23 +1,41 @@
 import { useEffect, useState } from "react";
 import { ago } from "../utils";
 import Votes from "./Votes";
-import { getUserByUsername } from "../api";
+import { deleteUserCommentById, getUserByUsername } from "../api";
 
-const CommentCard = ({ comment }) => {
-  const [user, setUser] = useState(null);
+const CommentCard = ({ comment, user, setComments }) => {
+  const [author, setAuthor] = useState(null);
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [hasDeleteError, setHasDeleteError] = useState(false);
 
   useEffect(() => {
     getUserByUsername(comment.author).then(({ user }) => {
-      setUser(user);
+      setAuthor(user);
     });
   }, []);
 
+  const handleDeleteComment = () => {
+    setHasDeleteError(false);
+    setIsDeleting(true);
+    deleteUserCommentById(comment.comment_id).then(() => {
+      setComments((prev) => {
+        return prev.filter((stateComment) => {
+          return stateComment.comment_id !== comment.comment_id;
+        });
+      });
+    }).catch(() => {
+      setHasDeleteError(true);
+    }).finally(() => {
+      setIsDeleting(false);
+    });
+  };
```

Comment:
Nice error handling here - could use a loading state for when the user has slow internet and presses the delete button to let them know deletion is in progress while they're waiting for the API response 🔍

Reply:
Thanks Katherine, great suggestion!